Add security blog post about fuzzing (#791)
fluxcd/website: content/en/blog/2022-02-22-security-fuzzing.md

We already have quite a good coverage of unit and end-to-end tests
across the controllers. Adding fuzzing to the mix will further extend
the scope of tests to scenarios and payloads not previously covered.
Together with the fuzzing that's already being done within the
Kubernetes repositories, e.g. `kubernetes`, `client-go` and
`apimachinery` we feel ever more confident in our code.

We are happy to share that since the 0.27 release of Flux all Flux
controllers and libraries are now tested by [Google's continuous
fuzzing for open source software](https://github.com/google/oss-fuzz).

## How we got here

When we [announced the results of the security
audit](/blog/2021-11-10-flux-security-audit/#flux-coming-to-oss-fuzz)
back in November, we already shared that the team at [ADA
Logics](https://adalogics.com/) had helped put together an initial
implementation of Fuzzing for some of the Flux controllers. In this
first inception three issues were already found (1x slice
out-of-bounds, 2x nil-dereference), and immediately fixed.
Naturally we were very interested in merging the fuzzing integration.

In order for us to fully land the fuzzers, we needed to make some

> architectural changes to the build process, especially for the
> controllers that have hard dependencies to `libgit2`, such as

Reviewer: ```suggestion
architectural changes to the build process, especially for the
controllers that rely on C bindings to `libgit2`, such as
```
